cclib/parser/gamessdatparser.py

```diff
@@ -118,129 +118,119 @@ def extract(self, inputfile, line):
+            self.metadata['num_nuclei'] = int(parts[6])
 
-        # DIPOLE       1.007144  1.336525  0.000000
+            # Continue extracting
+            
+            #   O    1    (CENTRE  1)   0.00000000  0.00000000  0.00000000  CHARGE =  8.0
+            #   H    2    (CENTRE  2)   1.87082873  0.00000000  0.00000000  CHARGE =  1.0
+            #   H    3    (CENTRE  3)  -0.51567032  1.79835585  0.00000000  CHARGE =  1.0
 
-        if line[1:7] == "DIPOLE":
-            line = next(inputfile).strip()
-            dipole_pattern = r"DIPOLE\s+([\d.-]+)\s+([\d.-]+)\s+([\d.-]+)"
-            match = re.search(dipole_pattern, line)
-            if match:
-                self.metadata["dipole"] = {
-                    "x": float(match.group(1)),
-                    "y": float(match.group(2)),
-                    "z": float(match.group(3))
-                }
-                
-        
-        # Extracting MP2 Energy Value
+            line = next(inputfile)
 
-        # MP2 NATURAL ORBITALS, E(MP2)=      -75.0022821133
-        
-        if "E(MP2)=" in line:
-            self.mpenergies = float(line.split()[-1])
+            atom_info = []
 
+            while 'CENTRE' in line:
 
-        # Extracting Gaussian
+                parts = line.split()
+                symbol = parts[0]
+                atom_number = int(parts[1])
+                centre_number = int(parts[3][:-1])
+                val1 = float(parts[4])
+                val2 = float(parts[5])
+                val3 = float(parts[6])
+                charge = float(parts[-1])
+
+                atom_info.append((symbol, atom_number, centre_number, val1, val2, val3, charge))
+
+                line = next(inputfile)
+
+            self.metadata['atom_info'] = atom_info
 
-        # ----- TOP OF INPUT FILE FOR BADER'S AIMPAC PROGRAM -----
-        # water                                                                           
-        # GAUSSIAN              7 MOL ORBITALS     21 PRIMITIVES        3 NUCLEI
-        #   O    1    (CENTRE  1)   0.00000000  0.00000000  0.00000000  CHARGE =  8.0
-        #   H    2    (CENTRE  2)   1.87082873  0.00000000  0.00000000  CHARGE =  1.0
-        #   H    3    (CENTRE  3)  -0.51567032  1.79835585  0.00000000  CHARGE =  1.0
 
         # Extracting Centre Assignments
 
         # CENTRE ASSIGNMENTS    1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  2  2  2  3  3
         # CENTRE ASSIGNMENTS    3
+        
+        self.metadata["centre_assignments"] = []
+
+        while line[0:18] == "CENTRE ASSIGNMENTS":
+            centre_assignments = re.findall(r'\d+', line)
+            self.metadata["centre_assignments"].extend(centre_assignments)
+            line = next(inputfile)
+
+        # Extracting Ttype Assignments
+
         # TYPE ASSIGNMENTS      1  1  1  1  1  1  2  2  2  3  3  3  4  4  4  1  1  1  1  1
         # TYPE ASSIGNMENTS      1
 
+        self.metadata["type_assignments"] = []
```

Comment:
The type assignment should be able to inform the construction of the 'gbasis' object. 

We'll need to chat about this a bit though as this would require some experience in atomic basis sets, but we should be able to plan it out together. For now this is just a comment and not yet a request for a specific change.